src/plugins/workflow_designer/src/WorkflowViewController.cpp

```diff
@@ -1975,8 +1994,14 @@ void WorkflowView::sl_saveScene() {
     }
     propertyEditor->commit();
     Task *t = new SaveWorkflowSceneTask(getSchema(), getMeta());
-    AppContext::getTaskScheduler()->registerTopLevelTask(t);
-    connect(t, SIGNAL(si_stateChanged()), SLOT(sl_onSceneSaved()));
+    if (saveImmideately) {
```

Comment:
Why can't you call the task the same way as before?

If the schema is asynchronously deleted it can happen with a regular scenario too: I click "save" and before the task is finished (it may be blocked in the task queue) I close the editor.

I see no problem to write the schema file from the main thread: we know that it is small. If you need this: extract the code in task::run into a static method utility and do not use the task in either case.